### PR TITLE
ACCT RELEASE 1612972949

### PR DESCRIPTION
**Pull request by themarcelor**
copying files from qa-acct.planx-pla.net to acct.bionimbus.org

### CODE REVIEW

File: acct.bionimbus.org/manifest.json [hunk 1]
```diff
@@ -8,24 +8,23 @@
     "autodeploy": "yes"
   },
   "versions": {
-    "ambassador": "quay.io/datawire/ambassador:1.4.2",
```
Comment (xritter1) [inline acct.bionimbus.org/manifest.json:11]: This line got deleted. Is this important though?

File: acct.bionimbus.org/etlMapping.yaml [hunk 2]
```diff
@@ -1,5 +1,5 @@
 mappings:
-  - name: acct_etl
+  - name: acct.bionimbus.org_subject
```
Comment (xritter1) [inline acct.bionimbus.org/etlMapping.yaml:2]: Why did this row change?

File: acct.bionimbus.org/etlMapping.yaml [hunk 3]
```diff
@@ -126,7 +126,7 @@ mappings:
         fn: count
     joining_props:
       - index: file
-        join_on: _subject_id
+        join_on: subject_id
```
Comment (xritter1) [inline acct.bionimbus.org/etlMapping.yaml:129]: Why did this row change?

File: acct.bionimbus.org/etlMapping.yaml [hunk 4]
```diff
@@ -137,7 +137,7 @@ mappings:
           - name: data_category
             src: data_type
             fn: set
-  - name: acct_file
+  - name: acct.bionimbus.org_file
```
Comment (xritter1) [inline acct.bionimbus.org/etlMapping.yaml:140]: Why did this row change?

File: acct.bionimbus.org/manifest.json [hunk 5]
```diff
@@ -36,9 +35,6 @@
   "peregrine": {
     "sidecar": "True"
   },
-  "jupyterhub": {
```
Comment (xritter1) [inline acct.bionimbus.org/manifest.json:39]: this needs to stay

File: acct.bionimbus.org/manifest.json [hunk 6]
```diff
@@ -49,12 +45,11 @@
     "logs_bucket": "logs-accountprod-gen3",
     "sync_from_dbgap": "False",
     "dispatcher_job_num": "10",
-    "useryaml_s3path": "s3://cdis-gen3-users/accountprod/user.yaml",
-    "netpolicy": "on"
```
Comment (xritter1) [inline acct.bionimbus.org/manifest.json:53]: why was this deleted? Is it important?

File: acct.bionimbus.org/manifest.json [hunk 7]
```diff
@@ -63,15 +58,15 @@
   "guppy": {
     "indices": [
       {
-        "index": "acct_etl",
-        "type": "subject"
+        "index": "acct.bionimbus.org_file",
+        "type": "file"
```
Comment (xritter1) [inline acct.bionimbus.org/manifest.json:62]: ?

File: acct.bionimbus.org/manifest.json [hunk 8]
```diff
@@ -63,15 +58,15 @@
...
       },
       {
-        "index": "acct_file",
-        "type": "file"
+        "index": "acct.bionimbus.org_subject",
+        "type": "subject"
       }
```
Comment (xritter1) [inline acct.bionimbus.org/manifest.json:67]: ?

File: acct.bionimbus.org/manifest.json [hunk 9]
```diff
@@ -63,15 +58,15 @@
...
     ],
-    "config_index": "acct_array-config",
+    "config_index": "acct.bionimbus.org_array-config",
     "auth_filter_field": "auth_resource_path"
```
Comment (xritter1) [inline acct.bionimbus.org/manifest.json:70]: ?

File: acct.bionimbus.org/portal/gitops.json [hunk 10]
```diff
@@ -44,32 +44,46 @@
       },
       "buttons": [
         {
-          "name": "Data Model",
-          "icon": "data-field-define",
-          "body": "The ACCOuNT Data Commons ingests and exports data that conform to a graph-like relational data model.",
+          "name": "Define Data Field",
```
Comment (xritter1) [inline acct.bionimbus.org/portal/gitops.json:47]: Nah. Needs to be "Data Model"

File: acct.bionimbus.org/portal/gitops.json [hunk 11]
```diff
@@ -44,32 +44,46 @@
...
+          "icon": "acct-planning",
+          "body": "The ACCOuNT Data Commons ingests and exports data that conform to a graph-like relational data model. Study the dictionary to send GraphQL queries or format your data submissions.",
```
Comment (xritter1) [inline acct.bionimbus.org/portal/gitops.json:49]: Delete additional sentence.

File: acct.bionimbus.org/portal/gitops.json [hunk 12]
```diff
@@ -79,253 +93,329 @@
...
           "link": "/query",
-          "color": "#a2a2a2",
+          "color": "#1a3057",
           "name": "query"
         },
         {
           "icon": "workspace",
-          "link": "/workspace",
-          "color": "#a2a2a2",
+          "link": "#hostname#workspace/",
```
Comment (xritter1) [inline acct.bionimbus.org/portal/gitops.json:113]: Nah. Needs to be the prod-version

File: acct.bionimbus.org/portal/gitops.json [hunk 13]
```diff
@@ -79,253 +93,329 @@
...
-          "icon": "upload",
-          "link": "/submission",
-          "name": "Browse Data"
-        },
-        {
-          "link": "https://gen3.org/resources/user/",
-          "name": "Gen3 Documentation"
-        }
-      ]
-    },
```
Comment (xritter1) [inline acct.bionimbus.org/portal/gitops.json:123]: this whole block needs to stay.

File: acct.bionimbus.org/portal/gitops.json [hunk 14]
```diff
@@ -79,253 +93,329 @@
...
     "login": {
-      "title": "ACCOuNT Data Commons",
+      "title": "ACCOuNT Data Commons (QA)",
```
Comment (xritter1) [inline acct.bionimbus.org/portal/gitops.json:126]: No.

File: acct.bionimbus.org/portal/gitops.json [hunk 15]
```diff
@@ -79,253 +93,329 @@
...
+      "#F3B44F"
+    ],
+    "categorical2Colors": [
+      "#1a3057",
+      "#C62C2E"
+    ],
     "footerLogos": [
       {
-        "src": "/custom/sponsors/gitops-sponsors/nimhd.jpg",
+        "src": "/src/img/sponsors/nimhd.png",
```
Comment (xritter1) [inline acct.bionimbus.org/portal/gitops.json:150]: No.

File: acct.bionimbus.org/portal/gitops.json [hunk 16]
```diff
@@ -79,253 +93,329 @@
...
       }
-    ],
-    "footer": {
-      "links": [
-        {
-          "text": "This project has been funded in whole or in part with Federal funds from the National Institute on Minority Health and Health Disparities, Grant Number U54 MD010723.",
-          "href": "https://precisionmedicine4all.com/about/"
-        }
-      ]
-    }
```
Comment (xritter1) [inline acct.bionimbus.org/portal/gitops.json:144]: No.

File: acct.bionimbus.org/portal/gitops.json [hunk 17]
```diff
@@ -79,253 +93,329 @@
...
+      "dropdowns": {}
+    }
+  ],
+  "dataExplorerConfig": {
+    "terraExportURL": null,
+    "sevenBridgesExportURL": null
+  },
+  "fileExplorerConfig": {
+    "terraExportURL": null,
+    "sevenBridgesExportURL": null
```
Comment (xritter1) [inline acct.bionimbus.org/portal/gitops.json:419]: I don't know where r 410-419 come from

**Review by xritter1 — COMMENTED** [review]
the image  acct.bionimbus.org/portal/gitops-sponsors/nimhd.png  should be already here: https://github.com/uc-cdis/cdis-manifest/tree/master/acct.bionimbus.org/portal/gitops-sponsors 
Why was it uploaded again? In QA the image was not displayed so we need to keep it as it was in prod..

File: acct.bionimbus.org/manifest.json (hunk 6 above)
Comment (xritter1) [reply acct.bionimbus.org/manifest.json:53]: i mean r53

File: acct.bionimbus.org/manifests/scaling/scaling.json [hunk 18]
```diff
@@ -0,0 +1,57 @@
...
+    "min": 1,
+    "max": 1,
+    "strategy": "auto"
+  },
+  "guppy": {
+    "min": 1,
+    "max": 1,
+    "strategy": "auto"
+  }
+}
```
Comment (xritter1) [inline acct.bionimbus.org/manifests/scaling/scaling.json:57]: this whole block is repetitive to what it's in the manifest I think

Comment (xritter1) [reply acct.bionimbus.org/manifests/scaling/scaling.json:57]: Should we remove it in the manifest?

File: acct.bionimbus.org/portal/gitops.json [hunk 19]
```diff
@@ -44,32 +44,46 @@
...
+          "color": "#1a3057"
         },
         {
           "name": "Analyze Data",
-          "icon": "data-analyze",
+          "icon": "acct-analyze",
           "body": "Analyze your selected cases using Jupyter Notebooks in our secure cloud environment",
-          "link": "/workspace",
-          "label": "Run analysis"
+          "link": "#hostname#workspace/",
```
Comment (xritter1) [inline acct.bionimbus.org/portal/gitops.json:74]: no

File: acct.bionimbus.org/manifest.json (hunk 1 above)
Comment (themarcelor) [reply acct.bionimbus.org/manifest.json:11]: Looks like workspaces were never enabled properly in `gitops-qa`, we can either:
1. Just push a commit to avoid this deletion
2. Introduce ambassador to `https://github.com/uc-cdis/gitops-qa/blob/master/qa-acct.planx-pla.net/manifest.json` and run the automation again to replicate all changes and create a new PR.

File: acct.bionimbus.org/etlMapping.yaml (hunk 2 above)
Comment (themarcelor) [reply acct.bionimbus.org/etlMapping.yaml:2]: This is working by design.
Our automation names the Elastic Search index according to the web address / hostname of the target environment.
This should be harmless.

File: acct.bionimbus.org/etlMapping.yaml (hunk 3 above)
Comment (themarcelor) [reply acct.bionimbus.org/etlMapping.yaml:129]: The automation is replicating whatever it finds in `gitops-qa/qa-acct.planx-pla.net`.
If ETL is working fine in QA, then I'd expect the same to work fine in PROD.
cc: @mpingram also holds ETL knowledge so perhaps we can help us +1 that.

File: acct.bionimbus.org/etlMapping.yaml (hunk 4 above)
Comment (themarcelor) [reply acct.bionimbus.org/etlMapping.yaml:140]: This is working by design.
Our automation names the Elastic Search index according to the web address / hostname of the target environment.
This should be harmless.

File: acct.bionimbus.org/manifest.json (hunk 5 above)
Comment (themarcelor) [reply acct.bionimbus.org/manifest.json]: Jupyterhub is an ancient component that is no longer being used.
This should be removed from all manifests.
cc'ing @atharvar28 to confirm (Hey Atharva, it is safe to remove that, right?)

File: acct.bionimbus.org/manifest.json (hunk 7 above)
Comment (themarcelor) [reply acct.bionimbus.org/manifest.json:65]: This is just matching the Elastic Search index name that was also set on etlMapping.yaml.
This is working by design.
Our automation names the Elastic Search index according to the web address / hostname of the target environment.
This should be harmless.

File: acct.bionimbus.org/manifest.json (hunk 6 above)
Comment (themarcelor) [reply acct.bionimbus.org/manifest.json:53]: It removed a comma (`,`) from the line with `useryaml_s3path `, which is fine.
But that next line `"netpolicy": "on"` should stay for security purposes.
(again, we can fix all that in `gitops-qa/qa-acct.planx-pla.net/manifest.json` and run the automation again to produce a new PR with all these corrections.

File: acct.bionimbus.org/manifest.json (hunk 8 above)
Comment (themarcelor) [reply acct.bionimbus.org/manifest.json:70]: This is just matching the Elastic Search index name that was also set on etlMapping.yaml.
This is working by design.
Our automation names the Elastic Search index according to the web address / hostname of the target environment.
This should be harmless.
--
💡 
Things will work fine as long as the names of the indices between the guppy config in `manifest.json` and `etlMapping.yaml` match.

File: acct.bionimbus.org/portal/gitops.json (hunk 10 above)
Comment (themarcelor) [reply acct.bionimbus.org/portal/gitops.json:47]: Again, the automation is just copying whatever it finds on `gitops-qa/qa-acct` so... if you plan to fix this in this PR, that's ok but please remember to also fix it in gitops-qa, otherwise, in the next release cycle, when this automation is executed again, the same bad text will be copied over.

File: acct.bionimbus.org/portal/gitops.json (hunk 11 above)
Comment (themarcelor) [reply acct.bionimbus.org/portal/gitops.json:49]: ? wat

File: acct.bionimbus.org/portal/gitops.json (hunk 12 above)
Comment (themarcelor) [reply acct.bionimbus.org/portal/gitops.json:113]: Again, the automation is supposed to copy whatever it finds on gitops-qa/qa-acct.
Did some bad copy happen here? Or did it ignore some json blocks from gitops-qa/qa-acct ?

File: acct.bionimbus.org/portal/gitops.json (hunk 13 above)
Comment (themarcelor) [reply acct.bionimbus.org/portal/gitops.json:123]: Again, the automation is supposed to copy whatever it finds on gitops-qa/qa-acct.
Did some bad copy happen here? Or did it ignore some json blocks from gitops-qa/qa-acct ?

File: acct.bionimbus.org/portal/gitops.json (hunk 14 above)
Comment (themarcelor) [reply acct.bionimbus.org/portal/gitops.json:126]: Oh, I see the problem here 😓 , let me prepare a fix so the automation will not replace that value in the next run.

File: acct.bionimbus.org/portal/gitops.json (hunk 15 above)
Comment (themarcelor) [reply acct.bionimbus.org/portal/gitops.json:150]: Oh, I see the problem here 😓 , let me prepare a fix so the automation will not replace that value in the next run.

File: acct.bionimbus.org/portal/gitops.json (hunk 16 above)
Comment (themarcelor) [reply acct.bionimbus.org/portal/gitops.json:144]: Oh, I see the problem here 😓 , let me prepare a fix so the automation will not replace that value in the next run.

File: acct.bionimbus.org/portal/gitops.json (hunk 17 above)
Comment (themarcelor) [reply acct.bionimbus.org/portal/gitops.json:419]: I'm guessing this is harmless (the automation just assumes these parameters are part of a common template and every environment should have them (which is a bad assumption).
But I think Windmill would just ignore these values if they are null..
let me cc: @mfshao to double-check. (just in case).

File: acct.bionimbus.org/manifests/scaling/scaling.json (hunk 18 above)
Comment (themarcelor) [reply acct.bionimbus.org/manifests/scaling/scaling.json:57]: The redundant config should be harmless.. 
But, ideally, we should tidy up `gitops-qa/qa-acct` and run this automation again.

File: acct.bionimbus.org/portal/gitops.json (hunk 19 above)
Comment (themarcelor) [reply acct.bionimbus.org/portal/gitops.json:74]: This should be `"link": "/workspace",`.
We should fix this in `gitops-qa/qa-acct` to fix the coinfig replication for subsequent release cycles.

File: acct.bionimbus.org/manifest.json (hunk 1 above)
Comment (xritter1) [reply acct.bionimbus.org/manifest.json:11]: right, we (@atharvar28 and I) removed the workspaces from QA because it wasn't needed. So it's a catch now... 
I added another commit to keep ambassador here in prod :)

File: acct.bionimbus.org/portal/gitops.json (hunk 13 above)
Comment (xritter1) [reply acct.bionimbus.org/portal/gitops.json:123]: the qa env is just not identical to prod 😭

File: acct.bionimbus.org/manifests/scaling/scaling.json (hunk 18 above)
Comment (xritter1) [reply acct.bionimbus.org/manifests/scaling/scaling.json:57]: this is only in the PROD manifest. Nothing like this is in the QA env afaik.
